ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/TimeModel.scala

```diff
@@ -53,11 +53,25 @@ object TimeModel {
       maxSkew = Duration.ofSeconds(30L),
     ).get
 
+  val resolution: Duration = Duration.ofNanos(1000L)
+
   def apply(avgTransactionLatency: Duration, minSkew: Duration, maxSkew: Duration): Try[TimeModel] =
     Try {
       require(!avgTransactionLatency.isNegative, "Negative average transaction latency")
       require(!minSkew.isNegative, "Negative min skew")
       require(!maxSkew.isNegative, "Negative max skew")
       new TimeModel(avgTransactionLatency, minSkew, maxSkew)
     }
+
+  private[state] def minRecordTime(ledgerTime: Instant, maxSkew: Duration): Instant =
+    ledgerTime.minus(maxSkew)
+
+  private[state] def maxRecordTime(ledgerTime: Instant, minSkew: Duration): Instant =
+    ledgerTime.plus(minSkew)
+
+  private def minLedgerTime(recordTime: Instant, minSkew: Duration): Instant =
+    recordTime.minus(minSkew)
+
+  private def maxLedgerTime(recordTime: Instant, maxSkew: Duration): Instant =
+    recordTime.plus(maxSkew)
 }
```

Comment:
I'd put those methods into the `TimeModel` class such that you don't have to manually pass the skew parameters.

Reply:
Agreed.